File: src/main/kotlin/lec02/Lec02.java
package lec02;

public class Lec02 {
    public static void main(String[] args) {

    }

    public boolean startWithA1(String str) {
        if (str == null)
            throw new IllegalArgumentException("null이 들어왔습니다.");
        return str.startsWith("A");
    }

    public Boolean startWithA2(String str) {
        if (str == null)
            return null;
        return str.startsWith("A");
    }

    public boolean startWithA3(String str) {
        if (str == null)
            return false;
        return str.startsWith("A");
    }
}

File: src/main/kotlin/lec02/Lec02.kt
package lec02

fun main() {
    var str: String? = "ABC"
     println(str?.length)

    str = null
    println(str?.length ?: 0)   // 엘비스 연산자 사용
}

fun startWithA1(str: String?): Boolean {
//    if (str == null)
//        throw IllegalArgumentException("null이 들어왔습니다.")
//    return str.startsWith("A")

    return str?.startsWith("A")
        ?: throw IllegalArgumentException("null이 들어왔습니다.") // Kotlin식 코드 변환
}

fun startWithA2(str: String?): Boolean? {
//    if (str == null)
//        return null
//    return str.startsWith("A")

    return str?.startsWith("A")
}

fun startWithA3(str: String?): Boolean {
//    if (str == null)
//        return false
//    return str.startsWith("A")

    return str?.startsWith("A") ?: false
}

fun startWith(str: String?): Boolean {
    return str!! .startsWith("A")
}